src/main/java/io/confluent/examples/streams/JsonToAvroExample.java

```
@@ -0,0 +1,133 @@
+    );
+    streams.start();
+
+    // Add shutdown hook to respond to SIGTERM and gracefully close Kafka Streams
+    Runtime.getRuntime().addShutdownHook(new Thread(streams::close));
+  }
+
+  static KafkaStreams buildJsonToAvroStream(final String bootstrapServers,
+      final String schemaRegistryUrl) {
+    final Properties streamsConfiguration = new Properties();
+
+    streamsConfiguration.put(StreamsConfig.APPLICATION_ID_CONFIG, "json-to-avro-stream-conversion");
+    streamsConfiguration.put(StreamsConfig.CLIENT_ID_CONFIG, "json-to-avro-stream-conversion-client");
+    streamsConfiguration.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers);
+    // Where to find the Confluent schema registry instance(s)
+    streamsConfiguration.put(AbstractKafkaAvroSerDeConfig.SCHEMA_REGISTRY_URL_CONFIG, schemaRegistryUrl);
+
+    streamsConfiguration.put(StreamsConfig.DEFAULT_VALUE_SERDE_CLASS_CONFIG, SpecificAvroSerde.class);
+    streamsConfiguration.put(StreamsConfig.DEFAULT_KEY_SERDE_CLASS_CONFIG, Serdes.String().getClass().getName());
+    streamsConfiguration.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+    streamsConfiguration.put(StreamsConfig.COMMIT_INTERVAL_MS_CONFIG, 10 * 1000);
+
+    final ObjectMapper objectMapper = new ObjectMapper();
+    final TypeReference<HashMap<String, String>> typeReference = new TypeReference<HashMap<String, String>>() {};
+
+    final StreamsBuilder builder = new StreamsBuilder();
+
+    // read the source stream
+    final KStream<String, String> jsonToAvroStream = builder.stream(JSON_SOURCE_TOPIC, Consumed.with(Serdes.String(), Serdes.String()));
+    jsonToAvroStream.mapValues(v -> {
+      WikiFeed wikiFeed = null;
+      try {
+        final Map<String, String> jsonMap = objectMapper.readValue(v, typeReference);
+        wikiFeed = new WikiFeed(jsonMap.get("user"), Boolean.valueOf(jsonMap.get("is_new")), jsonMap.get("content"));
+      } catch (final IOException e) {
+        e.printStackTrace();
```

It might be unsafe to ignore this exception. Should we recommend a more paranoid approach, like wrapping it with `RuntimeException` and rethrowing?

---

ack throwing `RuntimeException` now